front/lib/resources/datasource_resource.ts

```diff
@@ -0,0 +1,228 @@
+    const result: FindOptions<DataSourceResource["model"]> = {};
+
+    if (options?.includeEditedBy) {
+      result.include = [
+        {
+          model: User,
+          as: "editedByUser",
+        },
+      ];
+    }
+
+    if (options?.limit) {
+      result.limit = options.limit;
+    }
+
+    if (options?.order) {
+      result.order = options.order;
+    }
+
+    return result;
+  }
+
+  static async fetchByName(
+    auth: Authenticator,
+    name: string,
+    options?: Omit<FetchDataSourceOptions, "limit" | "order">
+  ): Promise<DataSourceResource | null> {
+    const owner = await auth.getNonNullableWorkspace();
+    const datasource = await this.model.findOne({
+      where: {
+        workspaceId: owner.id,
+        name,
+      },
+      ...this.getOptions(options),
+    });
+
+    if (!datasource) {
+      return null;
+    }
+    return new this(
+      DataSourceResource.model,
+      datasource.get(),
+      datasource.editedByUser?.get()
+    );
+  }
+
+  static async listByWorkspace(
+    auth: Authenticator,
+    options?: FetchDataSourceOptions
+  ): Promise<DataSourceResource[]> {
+    const owner = await auth.getNonNullableWorkspace();
+    const datasources = await this.model.findAll({
+      where: {
+        workspaceId: owner.id,
+      },
+      ...this.getOptions(options),
+    });
+
+    return datasources.map(
+      (datasource) =>
+        new this(
+          DataSourceResource.model,
+          datasource.get(),
+          datasource.editedByUser?.get()
+        )
+    );
+  }
+
+  static async listByWorkspaceIdAndNames(
+    workspace: LightWorkspaceType,
+    names: string[]
+  ): Promise<DataSourceResource[]> {
+    const datasources = await this.model.findAll({
+      where: {
+        workspaceId: workspace.id,
+        name: {
+          [Op.in]: names,
+        },
+      },
+    });
+
+    return datasources.map(
+      (datasource) => new this(DataSourceResource.model, datasource.get())
+    );
+  }
+
+  static async listByConnectorProvider(
+    auth: Authenticator,
+    connectorProvider: ConnectorProvider,
+    options?: FetchDataSourceOptions
+  ): Promise<DataSourceResource[]> {
+    const owner = await auth.getNonNullableWorkspace();
+    const datasources = await this.model.findAll({
+      where: {
+        workspaceId: owner.id,
+        connectorProvider,
+      },
+      ...this.getOptions(options),
+    });
+
+    return datasources.map(
+      (datasource) => new this(DataSourceResource.model, datasource.get())
+    );
+  }
+
+  async delete(
+    auth: Authenticator,
+    transaction?: Transaction
+  ): Promise<Result<undefined, Error>> {
+    await DataSourceViewResource.deleteForDataSource(auth, this);
+
+    try {
+      await this.model.destroy({
+        where: {
+          id: this.id,
+        },
+        transaction,
+      });
+
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+
+  async update(
```

Comment:
It's okay for the refactoring to have a global `update`, let's make sure that we refine it as we implement group and vault, to have dedicated method for each update case.